classes/SMTPMailingQueue.php

```diff
@@ -262,7 +263,13 @@ public static function storeMail($to, $subject, $message, $headers = '', $attach
 		$failures = 0;
 		$data = compact('to', 'subject', 'message', 'headers', 'attachments', 'time', 'failures');
 
-		$fileName = self::getUploadDir() . microtime(true) . '.json';
+		if ($invalid) {
+			$uploadType = 'invalid';
+		} else {
+			$uploadType = '';
+		}
+
+		$fileName = self::getUploadDir($uploadType) . uniqid('', true) . '.json';
```

Comment:
This seems perfect, did you try on a dev environment or should I ?
I find this method that could do the job too but I wonder if it is overkill for our situation : https://developer.wordpress.org/reference/functions/wp_unique_filename/

Reply:
No, I did not test it, I have no dev environment available for now. Could you please test it?
In this situation, I would not recommend using the `wp_unique_filename` function, as it generates a lot of I/O access to the file-system in case you have name conflicts what would often be the case in a queue situation.

Reply:
I totally agree with you. `wp_unique_filename` seems really poor in performance. It would be good to replace the corresponding call in "attachments" in the future. Do not hesitate to modify it in this PR if you like and have extra times.

Note : I just did multiple tests with uniqid and the fact that it returns hexa string is bad as (like you said) `glob` read will be randomly sorted. (Ex : 5f28503337197 > 5f28503daa59a) 

Searching into PHP code I discovered that they use `lcg_value` to create the `more_entropy`.
https://www.php.net/manual/fr/function.lcg-value.php
However it returns float between 0 and 1 which is not really easy to work with.

I discovered the `mt_rand` method which seems to be 4 time faster than lcg which could be good in our situation.
Example : $fileName = self::getUploadDir($uploadType) . microtime(true) . '.' . mt_rand(0, 999999999) . '.json';
Advantages : Fast, No hexa so ordering will be O.K
Disavtangages  :  A bit difficult to read the line

Funny : I found someone that used microtime as prefix for uniqId.
Example : $fileName = self::getUploadDir($uploadType) . uniqId(microtime(true), true) . '.json';
Advantages : Easier to read, Prefixed with microtime so ordering will be O.K
Disadvantages : Slower and it will compute two number based on microtime (one through microtime and one through uniqId)

My feeling go to mt_rand but I am interested  in yours.

Reply:
Good idea with `mt_rand`, but it is difficult to say whether it will be really faster as the string concatenation would be done in PHP space, while uniqid does it directly in C space... To be benchmarked if we really want to know!

Reply:
Moreover, I have made a small test with following php code:
```php
<?php
for($i=0;$i<20;$i++) {
    echo uniqid('', true), PHP_EOL;
}
?>
```
And I get this output:
```txt
5f28660a087069.25106599
5f28660a0a56f3.73905487
5f28660a0a6281.91305307
5f28660a0a7142.79283204
5f28660a0a7bc6.39788781
5f28660a0a8655.01612508
5f28660a0a9190.13684313
5f28660a0a9c45.91016585
5f28660a0aa6d8.10152380
5f28660a0ab653.40023411
5f28660a0ac0d9.52385070
5f28660a0acc38.14896330
5f28660a0ad6e8.85963056
5f28660a0ae134.29621229
5f28660a0aeba9.53575784
5f28660a0af630.52311256
5f28660a0b00d5.48194428
5f28660a0b0b90.25759707
5f28660a0b1724.93180737
5f28660a0b2197.27549732
```
We can observe that the first 13 digits (corresponding to `microtime(true)` in hexa) are keeping the alphabetical order.
I propose then to keep the `uniqid('', true)` approach for now.

Reply:
Totally agree with your point of view. Let's do this !

I will merge/version tomorrow.
Thanks a lot for your contribution @manandre.